### PR TITLE
[IL-318]: Improve Product Sorting on Homepage

### PR DESCRIPTION
**Pull request by Lightness322**
1. Task's name:
Improve Product Sorting on Homepage

2. Developer's full name:
Mikhail Dyubkov

3. Changes:
add new sort options
change dropdown styles
add default sort option

### CODE REVIEW

**Review by anksuunamun — CHANGES_REQUESTED** [review]

File: src/components/UI/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -46,14 +46,16 @@ const Dropdown = <T,>({
...
   }
 
-  const selectImageUrl = isOpen ? openSelect : closedSelect
-
   return (
-    <div ref={ref} className={'relative' + ' ' + (className ?? '')}>
+    <div
+      id="dropdown"
+      ref={ref}
+      className={'relative text-L text-primary' + ' ' + (className ?? '')}
```
Comment (freecree) [inline src/components/UI/Dropdown/Dropdown.tsx:53]: I think it's better to use twerge here. It allows to override styles of the component and it looks more readable
 ` className={twMerge('relative text-L text-primary', className)}`

**Review by freecree — CHANGES_REQUESTED** [review]

File: src/types/Dropdown.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 export type PropsDropdown<T> = {
   onChange: (option: IOption<T>) => void
-  options: IOption<T>[]
+  options: readonly IOption<T>[] | IOption<T>[]
```
[the remark this answers is not in the document]
Comment (Lightness322) [reply src/types/Dropdown.ts]: fixed

File: src/types/ProductSortParams.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-export interface IProductSortParams {
-  sortAttribute: string
-  sortDirection: 'asc' | 'desc'
-}
+import { sortOptions } from '@/constants/productSortOptions'
+
+export type ProductSortLabel = (typeof sortOptions)[number]['label']
```
[the remark this answers is not in the document]
Comment (Lightness322) [reply src/types/ProductSortParams.ts]: fixed

File: src/components/Product/ProductList/ProductList.tsx [hunk 4]
```diff
@@ -6,11 +6,15 @@ import ProductCard from '@/components/Product/ProductCard/ProductCard'
...
 import Dropdown from '@/components/UI/Dropdown/Dropdown'
 import ScrollUpBtn from '@/components/UI/Buttons/ScrollUpBtn/ScrollUpBtn'
-import { IProductSortParams } from '@/types/ProductSortParams'
 import { IOption } from '@/types/Dropdown'
 import ProductsFilterLabels from '@/components/Product/ProductsFilterLabels/ProductsFilterLabels'
 import { IProductFilterLabel } from '@/types/IProductFilterLabel'
 import { twMerge } from 'tailwind-merge'
+import { ProductSortLabel, ProductSortValue } from '@/types/ProductSortParams'
+
+function getDefaultSortOption(label: ProductSortLabel) {
```
[the remark this answers is not in the document]
Comment (Lightness322) [reply src/components/Product/ProductList/ProductList.tsx]: fixed

File: src/components/UI/Dropdown/Dropdown.tsx [hunk 5]
```diff
@@ -4,19 +4,19 @@ import { useOnClickOutside } from 'usehooks-ts'
 import { IOption, PropsDropdown } from '@/types/Dropdown'
 import { twMerge } from 'tailwind-merge'
 import Image from 'next/image'
-import openSelect from '../../../../public/open_select.svg'
-import closedSelect from '../../../../public/open_select_not_opened.svg'
+import selectImageUrl from '../../../../public/open_select.svg'
 import checkImageUrl from '../../../../public/check.svg'
```
[the remark this answers is not in the document]
Comment (Lightness322) [reply src/components/UI/Dropdown/Dropdown.tsx]: fixed

File: src/components/UI/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (Lightness322) [reply src/components/UI/Dropdown/Dropdown.tsx]: fixed

**Review by freecree — CHANGES_REQUESTED** [review]
image
Does such look of Dropdown in review section mathch the design? I think it should be with fixed width. If there is som uncertainty, I think it is better to ask designers

**Review by anksuunamun — DISMISSED** [review]

**Review by anksuunamun — APPROVED** [review]

**Comment by Lightness322** [comment]
> image Does such look of Dropdown in review section mathch the design? I think it should be with fixed width. If there is som uncertainty, I think it is better to ask designers

fixed

**Review by Sunagatov — APPROVED** [review]

**Pull request merged**